R/compare-opts.R

```diff
@@ -2,10 +2,13 @@ compare_opts <- function(...,
                          tolerance = NULL,
                          max_diffs = if (in_ci()) Inf else 10,
                          ignore_srcref = TRUE,
-                         ignore_attr = FALSE,
+                         ignore_attr = "waldo_opts",
                          ignore_encoding = TRUE,
+                         ignore_function_env = FALSE,
                          ignore_formula_env = FALSE,
-                         ignore_function_env = FALSE
+                         ignore_name_order = FALSE,
```

Comment:
Do you think this should default to `TRUE`?  (Will hopefully finish of this PR as a whole in the next week or two)

Reply:
Yes, name order shouldn't matter so a TRUE default would be sensible.  Arguments against it are:
 - it's a change
 - the existing user-visible differences are all `ignore_* = FALSE`.  Only encodings and srcrefs are ignored.

Reply:
Hmmmm, good points. I'm also a bit conflicted since waldo is designed for fairly low-level comparisons, and if the names are in different order subsetting by position will return different results. I think that all argues for leaving the default `FALSE`.